### PR TITLE
add density scaling performance test, ranging from 200, 400, and 600 VMis

### PR DESCRIPTION
**Pull request by marceloamaral**
Add a Prow job that runs a medium-scale density scaling performance test once a day by creating 200, 400, up to 600 VMIs in the performance cluster.

The results of this work will be used as the baseline performance of KubeVirt to evaluate a possible performance regression.

Also, I updated the density test to use golang images as the kubevirt build was failing because the bootstrap images don't have go.

This PR depends on kubevirt/kubevirt/pull/7060

/cc @fgimenez @rmohr @davidvossel @rthallisey

Signed-off-by: Marcelo Amaral <marcelo.amaral1@ibm.com>

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1148]: This preset is only required for windows and rhel tests

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1206]: This image tag is pretty old, could you please add a newer one? From https://quay.io/repository/kubevirtci/bootstrap?tab=tags, `v20220110-4643253` should be ok

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1161]: This is a lot of code, given that you need to propose `kubevirt-density-scale.yaml` to kubevirt/kubevirt, could you please also add this script under `hack`?

Comment (marceloamaral) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1131]: Done!

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (marceloamaral) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml]: Actually, building the kubevirt code was failing since bootstrap image does not have go installed.
I changed to golang image...

**Comment by marceloamaral** [comment]
@fgimenez I changed the PR as you requested.

Also, I updated the density test to use golang images as the kubevirt build was failing because the bootstrap images don't have go.

**Comment by dhiller** [comment]
@marceloamaral this is the first test of the job on the cluster: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-performance-cluster-scale-density-test/1484491434689040384

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1118]:
```suggestion
    preset-dind-enabled: "true"
    preset-kubevirtci-quay-credential: "true"
```

Comment (dhiller) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1118]: You would need to the quay credentials preset to have the env vars available in the job.

Comment (marceloamaral) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml]: Thanks @dhiller!
I updated it, could you please review it again?

**Review by dhiller — COMMENTED** [review]
/approve
/hold

Job lgtm, remove hold when everyone is ok with this.

**Review by dhiller — APPROVED** [review]

**Review by dhiller — APPROVED** [review]

**Pull request merged**
